From 5c78cc31e908d46c7c5e0b4505cc8aaa182bbd0b Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:39:32 -0400
Subject: [alsa] hda: Add mic fixup for Gigabyte BXBT-2807

Message-id: <140972823573-398-patchset-jkysela@redhat.com>
Patchwork-id: 91094
O-Subject: [RHEL 7.1 PATCH 397/442] ALSA: hda - Add mic fixup for Gigabyte BXBT-2807
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The Gigabyte BRIX BXBT-2707 is a mini-PC with Realtek ALC283 HDA,
exposing a single headset jack.

However, the default pin config information only suggests
that one pin is connected: a HP out jack (pin 0x21, default
config 0x04211010).

The microphone input is behind pin 0x19, which has default
config 0x411111f0 (i.e. unused), so it does not show up in
userspace, and no microphone input is possible via the headset.

Override the pin config so that the headset mic can be used.

[rearranged the fixup entry position by tiwai]

Signed-off-by: Daniel Drake <drake@endlessm.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 9dc12862da9d56ef4da646ba540c4f58b78738fc)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 6f204ba..124272b 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -4358,6 +4358,7 @@ enum {
  ALC255_FIXUP_HEADSET_MODE_NO_HP_MIC,
  ALC293_FIXUP_DELL1_MIC_NO_PRESENCE,
  ALC292_FIXUP_TPT440_DOCK,
+ ALC283_FIXUP_BXBT2807_MIC,
 };
 
 static const struct hda_fixup alc269_fixups[] = {
@@ -4792,6 +4793,13 @@ static const struct hda_fixup alc269_fixups[] = {
   .chained = true,
   .chain_id = ALC269_FIXUP_LIMIT_INT_MIC_BOOST
  },
+ [ALC283_FIXUP_BXBT2807_MIC] = {
+  .type = HDA_FIXUP_PINS,
+  .v.pins = (const struct hda_pintbl[]) {
+   { 0x19, 0x04a110f0 },
+   { },
+  },
+ },
 };
 
 static const struct snd_pci_quirk alc269_fixup_tbl[] = {
@@ -4987,6 +4995,7 @@ static const struct snd_pci_quirk alc269_fixup_tbl[] = {
  SND_PCI_QUIRK_VENDOR(0x104d, "Sony VAIO", ALC269_FIXUP_SONY_VAIO),
  SND_PCI_QUIRK(0x10cf, 0x1475, "Lifebook", ALC269_FIXUP_LIFEBOOK),
  SND_PCI_QUIRK(0x10cf, 0x1845, "Lifebook U904", ALC269_FIXUP_LIFEBOOK_EXTMIC),
+ SND_PCI_QUIRK(0x1458, 0xfa53, "Gigabyte BXBT-2807", ALC283_FIXUP_BXBT2807_MIC),
  SND_PCI_QUIRK(0x17aa, 0x20f2, "Thinkpad SL410/510", ALC269_FIXUP_SKU_IGNORE),
  SND_PCI_QUIRK(0x17aa, 0x215e, "Thinkpad L512", ALC269_FIXUP_SKU_IGNORE),
  SND_PCI_QUIRK(0x17aa, 0x21b8, "Thinkpad Edge 14", ALC269_FIXUP_SKU_IGNORE),
-- 
1.7.1